boards/posix/native_posix/Kconfig.board

```diff
@@ -1,13 +1,24 @@
 # SPDX-License-Identifier: Apache-2.0
 
-config BOARD_NATIVE_POSIX
-	bool "Native POSIX"
+config BOARD_NATIVE_POSIX_32BIT
```

Comment:
This is kind of ok. But I'm thinking it may confuse people, as one will now compile targetting native_posix or native_posix_64, but not native_posix_32. (mind you, I certainly think we should be "backwards compatible" and have "native_posix" still existing).
Really BOARD_NATIVE_POSIX_32BIT is nothing more than selecting BOARD_NATIVE_POSIX, or?

Reply:
That proposal of mine above would probably mean replacing every current instance in this PR branch of BOARD_NATIVE_POSIX with something else (say NATIVE_POSIX_COMMON), and then leaving in this file on BOARD_NATIVE_POSIX and BOARD_NATIVE_POSIX_64. It may not be worth the trouble.

Reply:
I'm fine with either one. `NATIVE_POSIX` matches the `-DBOARD` flag, and `NATIVE_POSIX_32` makes it clear that it's the 32-bit version. Different pros. Not sure which one is nicer.

Unrelated, but I think our board selection thing is pretty confusing. Since a bunch of other, non-Kconfig-related stuff is set when you select the board on the command line, it couldn't ever work to change th board within Kconfig. Maybe it shouldn't even have a prompt.